### PR TITLE
Add description of GLOBAL level of tidb_txn_mode variable

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What is changed, added or deleted? 

[This PR](https://github.com/pingcap/tidb/pull/12041) makes the `tidb_txn_mode` variable supports global scope. So the corresponding document should also be updated. But since the change is not released yet, and the next release that contains this change is 3.0.4, so **this PR should be merged after releasing 3.0.4**.

### What is the related PR or file link(s)? 

N/A

### Which version does your change affect? 

3.0, dev

### CODE REVIEW

**Comment by coocood** [comment]
We can merge dev first.

**Comment by MyonKeminta** [comment]
If so we need submit 2 PRs to update it twice. I'm afraid it makes work too fragmentary.

File: dev/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 1]
```diff
@@ -480,11 +480,11 @@ set tidb_query_log_max_len = 20
 
 ### tidb_txn_mode
 
-作用域：SESSION
+作用域：SESSION （自 3.0.4 起支持 GLOBAL）
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
作用域：SESSION（自 TiDB 3.0.4 起支持 GLOBAL）
```

File: dev/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 2]
```diff
@@ -480,11 +480,11 @@ set tidb_query_log_max_len = 20
...
 
 默认值：""
 
-这个变量用于设置当前 session 的事务模式，默认是乐观锁模式。 TiDB 3.0 加入了悲观锁模式（实验性）。将 `tidb_txn_mode` 设置为 `'pessimistic'` 后，这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。更多关于悲观锁的细节，可以参考 [TiDB 悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)。
+这个变量用于设置事务模式，默认是乐观锁模式。 TiDB 3.0 加入了悲观锁模式（实验性）。将 `tidb_txn_mode` 设置为 `'pessimistic'` 后，这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。自 3.0.4 起，该变量也支持 GLOBAL 作用域，用于设定全局的事务模式。更多关于悲观锁的细节，可以参考 [TiDB 悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)。自 3.0.4 起，该变量也支持 GLOBAL 作用域，设定全局的事务模式。
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
这个变量用于设置事务模式，默认是乐观锁模式。TiDB 3.0 加入了悲观锁模式（实验性）。将 `tidb_txn_mode` 设置为 `'pessimistic'` 后，这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。更多关于悲观锁的细节，可以参考 [TiDB 悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)。

自 TiDB 3.0.4 起，该变量也支持 GLOBAL 作用域，用于设定全局的事务模式。
```

**Comment by MyonKeminta** [comment]
@yikeke @coocood PTAL

File: v3.0/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 3]
```diff
@@ -490,11 +490,13 @@ set tidb_query_log_max_len = 20;
 
 ### tidb_txn_mode
 
-作用域：SESSION
+作用域：SESSION（自 TiDB 3.0.4 起支持 GLOBAL）
 
 默认值：""
 
-这个变量用于设置当前 session 的事务模式，默认是乐观锁模式。 TiDB 3.0 加入了悲观锁模式（实验性）。将 `tidb_txn_mode` 设置为 `'pessimistic'` 后，这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。更多关于悲观锁的细节，可以参考 [TiDB 悲观事务模式](/v3.0/reference/transactions/transaction-pessimistic.md)。
+这个变量用于设置事务模式，默认是乐观锁模式。TiDB 3.0 加入了悲观锁模式（实验性）。将 `tidb_txn_mode` 设置为 `'pessimistic'` 后，这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。更多关于悲观锁的细节，可以参考 [TiDB 悲观事务模式](/dev/reference/transactions/transaction-pessimistic.md)。
```
Comment (yikeke) [inline v3.0/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
这个变量用于设置事务模式，默认是乐观锁模式。TiDB 3.0 加入了悲观锁模式（实验性）。将 `tidb_txn_mode` 设置为 `'pessimistic'` 后，这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。更多关于悲观锁的细节，可以参考 [TiDB 悲观事务模式](/v3.0/reference/transactions/transaction-pessimistic.md)。
```
This change will fix the CI.

**Pull request merged**